spring-integration-core/src/main/java/org/springframework/integration/support/management/DefaultMessageChannelMetrics.java

```diff
@@ -114,7 +152,11 @@ public void afterSend(MetricsContext context, boolean result) {
 			if (isFullStatsEnabled()) {
 				long now = System.nanoTime();
```

Comment:
A consequence of adding `TimerFacade` -- it would be preferable to use Micrometer's `Timer#record(Runnable)` or variants so that the provided `Clock` is honored. `System.nanoTime` bypasses the provided clock.

Reply:
`record(Runnable)` would require some major changes which we can't really do in a point release, and in any case would be tough to do by Wednesday, which is our release date.

We can rework in 5.1; if this is a show-stopper we will have to rethink.

Reply:
Not a show-stopper, just something to keep in mind going forward.